src/main/java/com/katalon/jenkins/plugin/ExecuteKatalonTestOpsPlan.java

```diff
@@ -175,10 +176,13 @@ private String getApiKey(String credentialsId) {
       return credentials == null ? null : credentials.getSecret().getPlainText();
     }
 
+    @RequirePOST
     public FormValidation doTestConnection(@QueryParameter("serverUrl") final String url,
                                            @QueryParameter("credentialsId") final String credentialsId) {
+      Jenkins.getActiveInstance().checkPermission(Item.CONFIGURE);
```

Comment:
This is the wrong permission check: You're checking a permission scoped to items globally.

1. You will prevent people from using this when they have permission to configure a specific plugin.
2. You will allow people who have the permission globally, even when all jobs disable permission inheritance.

See the two examples at https://www.jenkins.io/doc/developer/security/form-validation/#implementing-form-validation and choose the behavior you need for your plugin.

Reply:
Updated. Please review again